src/openforms/registrations/contrib/objects_api/config.py

```diff
@@ -453,20 +458,40 @@ def _validate_catalogue_and_document_types(attrs: RegistrationOptions) -> None:
         "informatieobjecttype_attachment",
     ):
         url = attrs.get(field)
-        if url:
-            warnings.warn(
-                "URL references to document types are deprecated and will be remove in "
-                "Open Forms 3.0",
-                DeprecationWarning,
-            )
-        if url and url not in informatieobjecttypen_urls:
-            err_tpl = (
-                _("The provided {field} does not exist in the Catalogi API.")
-                if catalogus is None
-                else _("The provided {field} does not exist in the selected catalogue.")
-            )
-            err_msg = err_tpl.format(field=field)
-            raise serializers.ValidationError({field: err_msg}, code="not-found")
+        if not url:
+            continue
+
+        warnings.warn(
+            "URL references to document types are deprecated and will be remove in "
+            "Open Forms 3.0",
+            DeprecationWarning,
+        )
+
+        err_tpl = (
+            _("The provided {field} does not exist in the Catalogi API.")
+            if catalogus is None
+            else _("The provided {field} does not exist in the selected catalogue.")
+        )
+        err_msg = err_tpl.format(field=field)
+
+        if informatieobjecttypen_urls:
+            if url not in informatieobjecttypen_urls:
+                raise serializers.ValidationError({field: err_msg}, code="not-found")
+        else:
+            assert api_group.catalogi_service
+            iotypen_endpoint = (
+                furl(api_group.catalogi_service.api_root) / "informatieobjecttypen"
+            ).url
+
+            if not url.startswith(iotypen_endpoint):
+                raise serializers.ValidationError({field: err_msg}, code="not-found")
+            with get_catalogi_client(api_group) as client:
```

Comment:
I think we have to indent the whole code block to fit the context manager on line 396, since now you're opening and closing HTTP connections for each field. This would benefit from sharing the same connection pool as the initial Catalogi API interaction.